sdk/cosmos/src/clients/cosmos_client.rs

```diff
@@ -158,28 +163,80 @@ impl CosmosClient {
+                        return Some((Err(err.into()), State::Done));
+                    }
+                }
+            };
+        }
 
-        options.decorate_request(&mut request).await?;
-        let response = self
-            .pipeline()
-            .send(&mut pipeline_context, &mut request)
-            .await?
-            .validate(http::StatusCode::OK)
-            .await?;
+        #[derive(Debug, Clone, PartialEq)]
+        enum State {
+            Init,
+            Continuation(String),
+            Done,
+        }
+
+        unfold(State::Init, move |state: State| {
+            let this = self.clone();
+            let ctx = ctx.clone();
+            let options = options.clone();
+            async move {
+                let response = match state {
+                    State::Init => {
+                        let mut request = this.prepare_request_pipeline("dbs", http::Method::GET);
+                        let mut pipeline_context =
+                            PipelineContext::new(ctx.clone(), ResourceType::Databases.into());
+
+                        r#try!(options.decorate_request(&mut request).await);
+                        let response = r#try!(
+                            this.pipeline()
+                                .send(&mut pipeline_context, &mut request)
+                                .await
+                        );
+                        let response = r#try!(response.validate(http::StatusCode::OK).await);
 
-        Ok(ListDatabasesResponse::try_from(response).await?)
+                        ListDatabasesResponse::try_from(response).await
+                    }
+                    State::Continuation(continuation_token) => {
+                        let continuation = Continuation::new(continuation_token.as_str());
+                        let mut request = this.prepare_request_pipeline("dbs", http::Method::GET);
+                        let mut pipeline_context =
+                            PipelineContext::new(ctx.clone(), ResourceType::Databases.into());
+
+                        r#try!(options.decorate_request(&mut request).await);
+                        r#try!(continuation.add_as_header2(&mut request));
+                        let response = r#try!(
+                            this.pipeline()
+                                .send(&mut pipeline_context, &mut request)
+                                .await
+                        );
+                        let response = r#try!(response.validate(http::StatusCode::OK).await);
+                        ListDatabasesResponse::try_from(response).await
+                    }
+                    State::Done => return None,
+                };
+
+                let response = r#try!(response);
+
+                let continuation_token = response
+                    .continuation_token
+                    .clone()
+                    .map(|ct| State::Continuation(ct))?;
+
+                Some((Ok(response), continuation_token))
```

Comment:
```suggestion
                let next_state = response
                    .continuation_token
                    .clone() 
                    .map(|ct| State::Continuation(ct)) 
                    .unwrap_or_else(|| State::Done);           

                Some((Ok(response), next_state))
```

The above code incorrectly mapped the absence of `continuation_token` to the absence of data (which caused the bug). A simple `.unwrap_or_else(|| State::Done)` should be enough!

As an aside, I suggest renaming the binding from `continuation_token` to the more appropriate `next_state`.